lxd/device/gpu.go

```diff
@@ -0,0 +1,644 @@
+package device
+
+import (
+	"encoding/csv"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/jaypipes/pcidb"
+	"golang.org/x/sys/unix"
+
+	"github.com/lxc/lxd/lxd/device/config"
+	"github.com/lxc/lxd/lxd/instance"
+	"github.com/lxc/lxd/shared"
+)
+
+type gpu struct {
+	deviceCommon
+}
+
+// /dev/dri/card0. If we detect that vendor == nvidia, then nvidia will contain
+// the corresponding nvidia car, e.g. {/dev/dri/card1 to /dev/nvidia1}.
+type gpuDevice struct {
```

Comment:
Would be good to see if we can get rid of all that in favor of `resources.GetGPU()`.
Not a blocker, but would be a good follow-up as we may be able to remove a significant chunk of the code.

Same may be true for infiniband with `resources.GetNetwork()` as it now reports the infiniband devices too.